src/v0/destinations/braze/transform.js

```diff
@@ -308,10 +312,82 @@ function processTrackEvent(messageType, message, destination, mappingJson) {
   payload = addMandatoryEventProperties(payload, message);
   payload.properties = properties;
 
+  try {
+    // add,update,remove
+    if (destination.Config.enableNestedArrayOperations) {
+      Object.keys(properties)
+        .filter(k => Array.isArray(properties[`${k}`]))
+        .forEach(key => {
+          // if not specified, send as create attribute
+          if (properties.nestedOperationType === nestedOperationTypes.CREATE) {
+            attributePayload[key] = properties[key];
+          } else if (
+            supportedOperationTypes.includes(properties.nestedOperationType)
+          ) {
+            attributePayload[key] = {};
+            const opsResultArray = [];
+            if (
+              properties.nestedOperationType === nestedOperationTypes.UPDATE
+            ) {
+              for (let i = 0; i < properties[key].length; i += 1) {
+                const myObj = {};
+                Object.keys(properties[key][i]).forEach(subKey => {
+                  myObj[`$${subKey}`] = properties[key][i][subKey];
+                });
+                opsResultArray.push(myObj);
+              }
+
+              // eslint-disable-next-line no-underscore-dangle
+              attributePayload._merge_objects = isDefinedAndNotNull(
+                properties.mergeObjectsUpdateOperation
+              )
+                ? properties.mergeObjectsUpdateOperation
+                : false;
+              attributePayload[key][
+                `$${properties.nestedOperationType}`
+              ] = opsResultArray;
+            } else if (
+              properties.nestedOperationType === nestedOperationTypes.REMOVE
+            ) {
+              for (let i = 0; i < properties[key].length; i += 1) {
+                const myObj = {};
+                Object.keys(properties[key][i]).forEach(subKey => {
+                  myObj[`$${subKey}`] = properties[key][i][subKey];
+                });
+                opsResultArray.push(myObj);
+              }
+              attributePayload[key][
+                `$${properties.nestedOperationType}`
+              ] = opsResultArray;
+            } else {
+              // add case
+              attributePayload[key][`$${properties.nestedOperationType}`] =
+                properties[key];
+            }
+          }
+        });
+    }
+  } catch (exp) {
+    logger.info("Failure occured during nested array operations", exp);
+  }
+
   payload = setExternalIdOrAliasObject(payload, message);
   if (payload) {
     requestJson.events = [payload];
   }
+
+  // update payload as per https://www.braze.com/docs/user_guide/data_and_analytics/custom_data/custom_attributes/array_of_objects/#api-request-body
+  if (
+    destination.Config.enableNestedArrayOperations &&
+    isDefinedAndNotNull(properties.nestedOperationType)
+  ) {
+    delete requestJson.events;
```

Comment:
We should ideally avoid explicit deletion of object properties in our transformation code. Can we utilize something like [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) if possible? CC @ItsSudip 

|non blocking|

Reply:
As this is final payload after transformation, I think we can delete un-required keys from it. These props are req for other flow, not for this change.